tests/plugins/button/buttonicon.js

```diff
@@ -0,0 +1,225 @@
+/* bender-tags: editor */
+/* bender-ckeditor-plugins: button,toolbar */
+
+var editorCounter = 0,
+	originalBasePath = null,
+	isDev = CKEDITOR.version === '%VERSION%',
+	isHidpi;
+
+function resolutionStandard() {
+	CKEDITOR.env.hidpi = false;
+}
+
+function resolutionHidpi() {
+	CKEDITOR.env.hidpi = true;
+}
+
+function assertIcon( editorConfig, btnName, iconPath ) {
+	editorCounter++;
+	bender.editorBot.create( {
+		name: 'editor' + editorCounter,
+		config: editorConfig
+	}, function( bot ) {
+		var btn = bot.editor.ui.get( btnName ),
+			btnEl = CKEDITOR.document.getById( btn._.id ),
+			btnCss = CKEDITOR.tools.parseCssText( btnEl.findOne( '.cke_button_icon' ).getAttribute( 'style' ), true );
+
+		assert.isMatching( iconPath, btnCss[ 'background-image' ] );
+	} );
+}
+
+bender.test( {
+	init: function() {
+		isHidpi = CKEDITOR.env.hidpi;
+	},
+
+	tearDown: function() {
+		// Restore global values modified by tests.
+		CKEDITOR.env.hidpi = isHidpi;
+		if ( originalBasePath ) {
+			CKEDITOR.basePath = originalBasePath;
+			originalBasePath = null;
+		}
+	},
+
+	'test default button icon': function() {
+		resolutionStandard();
+		assertIcon(
+			{ extraPlugins: 'link' },
+			'Link',
+			isDev ? /plugins\/link\/icons\/link\.png/gi : /plugins\/icons\.png/gi
+		);
+	},
+
+	'test default button icon (hidpi)': function() {
```

Comment:
Hm, personally I'd simplify tests and encapsulate switching between standard/hidpi inside `createIconTest` or something like that.

Reply:
This is problematic due to a fact how icons are registered. Icon is registered on plugin init so when plugin is loaded on standard resolution standard icons are initialized. Then switching to HiDPI and creating new editor (with same plugin) would not reregister the icons so the standard one is still in use.

This is the reason standard an HiDPI tests use different plugins. This is also a reason why generating two tests (standard one and HiDPI) from one editor config doesn't make sense thus auto generation doesn't make much sense (since you have to pass two different set of settings).

I refactored the tests creation to be more automated, however still standard and HiDPI tests are not generated form one set of settings.